pkg/kubelet/kubelet_node_status.go

```diff
@@ -341,23 +341,32 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 }
 
 func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
+	// Note: avoid blindly overwriting the capacity in case opaque
+	//       resources are being advertised.
+	if node.Status.Capacity == nil {
+		node.Status.Capacity = api.ResourceList{}
```

Comment:
Can you elaborate? Previously this function unconditionally clobbered `node.Status.Capacity`.